### ISSUE TITLE
Numeric marshaled values are rounded to the nearest integer

### ISSUE THREAD

**Issue by coddingtonbear** [ralphbean/taskw#136]
Today while triaging bug tickets on Inthe.AM, I found https://github.com/coddingtonbear/inthe.am/issues/269 describing a problem in which tasks were mis-ordered because the urgency value returned by the API was unexpectedly rounded.  A little digging revealed that this is actually some kind of problem in Taskw's marshalling:

Unmarshalled:

```
>>> tws = TaskWarriorShellout('./.taskrc', marshal=False)
>>> tws.filter_tasks({'description': 'Double'})[0]
{'id': 15, 'description': 'Double check that AIRSHIP-4Z1 fix worked: https://github.com/urbanairship/airship/pull/8526 worked?', 'entry': '20190419T173354Z', 'intheamtrelloboardid': '57fc50d53515d1ba1abe38a3', 'intheamtrelloid': '5cba067d2b11711351568815', 'intheamtrellolistid': '57fc51aa7b70585afcb3da79', 'intheamtrellolistname': 'This Week', 'intheamtrellourl': 'https://trello.com/c/QSTpkMCS/2220-double-check-that-airship-4z1-fix-worked-https-githubcom-urbanairship-airship-pull-8526-worked', 'modified': '20190419T173948Z', 'status': 'pending', 'tags': ['sky', 'work'], 'uuid': '54f30f75-1bbb-4023-9405-44383939f99c', 'urgency': 3.15}
```

Marshalled: 

```
>>> tws = TaskWarriorShellout('./.taskrc', marshal=True)
>>> tws.filter_tasks({'description': 'Double'})[0]
{'id': 15, 'description': 'Double check that AIRSHIP-4Z1 fix worked: https://github.com/urbanairship/airship/pull/8526 worked?', 'entry': datetime.datetime(2019, 4, 19, 17, 33, 54, tzinfo=tzlocal()), 'intheamtrelloboardid': '57fc50d53515d1ba1abe38a3', 'intheamtrelloid': '5cba067d2b11711351568815', 'intheamtrellolistid': '57fc51aa7b70585afcb3da79', 'intheamtrellolistname': 'This Week', 'intheamtrellourl': 'https://trello.com/c/QSTpkMCS/2220-double-check-that-airship-4z1-fix-worked-https-githubcom-urbanairship-airship-pull-8526-worked', 'modified': datetime.datetime(2019, 4, 19, 17, 39, 48, tzinfo=tzlocal()), 'status': 'pending', 'tags': ['sky', 'work'], 'uuid': UUID('54f30f75-1bbb-4023-9405-44383939f99c'), 'urgency': 3}
```

This is undoubtedly somehow my fault, of course :-), whether this is filed under Inthe.AM or not.

**Comment by sjbronner** [comment]
I just ran across this while debugging the same issue in telch. I believe the problem is in `fields/numeric.py` in the function `NumericField.deserialize`. Deserialization as an integer is preferred there over deserialization as a float, regardless of the presence of decimal digits or a decimal point. This basically ensures that all numbers output from `task` appear rounded in python.

In my opinion, a somewhat more complex check would be in order: `if float(value) != int(value): return float(value)`, otherwise `if '.' in value: return float(value)`, otherwise `return int(value)`. The conditions for `None` and a non-numeric `value` should be kept as they are.

**Comment by sjbronner** [comment]
It turns out the following patch is enough to alleviate this specific problem:
```
--- /usr/lib/python3/dist-packages/taskw/fields/numeric.py.orig 2015-01-19 16:12:46.000000000 +0100
+++ /usr/lib/python3/dist-packages/taskw/fields/numeric.py      2020-12-06 23:34:32.081430072 +0100
@@ -11,7 +11,7 @@
         return value

     def deserialize(self, value):
-        if value is None:
+        if value is None or isinstance(value, (int, float)):
             return value
         try:
             return int(value)
```

**Comment by coddingtonbear** [comment]
Merged a fix for this via the changes in #124
